src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,234 @@
+
+  const allCenturySelection = () => {
+    setactiveCenturies([]);
+  };
+
+  const handleSort = (column: string) => {
+    setsortField(column);
+    setClickCount(clickCount + 1);
+  };
+
+  const resetEveryThing = () => {
+    setactiveCenturies([]);
+    setQuery('');
+    sexFilterHandler(FilterType.All);
+    localStorage.setItem('query', '');
+  };
+
+  const setCurrentQuery = (currentQuery: string) => {
+    localStorage.setItem('query', currentQuery);
+
+    setQuery(currentQuery);
+  };
+
+  const handleCenturySelection = (century: number) => {
+    const includesCenturie = activeCenturies.includes(century);
+
+    if (includesCenturie && activeCenturies.length === 5) {
+      const filtered = activeCenturies.filter((number) => {
+        return number === century;
+      });
+
+      setactiveCenturies(filtered);
+    } else if (!includesCenturie
+    && activeCenturies.length !== 5) {
+      setactiveCenturies([...activeCenturies, century]);
+    } else if (includesCenturie) {
+      const filtered = activeCenturies.filter((number) => {
+        return number !== century;
+      });
+
+      setactiveCenturies(filtered);
+    }
+  };
+
+  const updatedPeople = people.map((child) => {
+    if (!child.motherName && !child.fatherName) {
+      return {
+        ...child,
+        motherName: '-',
+        fatherName: '-',
+      };
+    }
+
+    if (!child.fatherName) {
+      return {
+        ...child,
+        fatherName: '-',
+      };
+    }
+
+    if (!child.motherName) {
+      return {
+        ...child,
+        motherName: '-',
+      };
+    }
+
+    const father = people.find((parent) => parent.name === child.fatherName);
+    const mother = people.find((parent) => parent.name === child.motherName);
+
+    return {
+      ...child,
+      father,
+      mother,
+    };
+  });
+
+  const filteredPeople: NewPerson[] = useMemo(
+    () => updatedPeople.filter((person) => {
+      const functionality
+       = person.name.toLowerCase().includes(query.toLowerCase().trim());
+      const centurie = Math.floor(person.born / 100);
+
+      switch (sexFilter) {
+        case FilterType.All:
+        default:
+          return activeCenturies.length === 0
+            ? functionality
+            : functionality && activeCenturies.includes(centurie);
+        case FilterType.Male:
+          return (
+            activeCenturies.length === 0
+              ? (person.sex === 'm'
+              && functionality)
+              : (person.sex === 'm'
+              && functionality
+              && activeCenturies.includes(centurie))
+          );
+        case FilterType.Female:
+          return (
+            activeCenturies.length === 0
+              ? (person.sex === 'f'
+              && functionality)
+              : (person.sex === 'f'
+              && functionality
+              && activeCenturies.includes(centurie))
+          );
+      }
+    }),
+    [sexFilter, query, loading, activeCenturies, fetchPeople],
```

Comment:
Remove 'fetchPeople' from the dependency array as it is not being used inside the useMemo.

Reply:
Didn't fix